src/timestamp9.c

```diff
@@ -291,9 +299,9 @@ long long parse_fractional_ratio(const char* str, size_t len, bool* fractional_v
 	size_t i = 0;
 	*fractional_valid = false;
 
-	while (i < len)
+	while (i <= len)
 	{
-		if (count && (str[i] == ' ' || str[i] == '+' || str[i] == '-' || str[i] == 'Z'))
+		if (count && (str[i] == ' ' || str[i] == '+' || str[i] == '-' || str[i] == 'Z' || str[i] == '\0'))
```

Comment:
this looks a bit dubious. if `!count` we won't break out of the loop even if we reach the null char

Reply:
The length of the buffer is calculated by `len = strlen(str);`, so the loop will end if we reach `\0`. 

Maybe refactoring it to `if ((count && (str[i] == ' ' || str[i] == '+' || str[i] == '-' || str[i] == 'Z')) || str[i] == '\0')` is better?

Reply:
ah you're right, I missed that the loop will still break for the regular `while` condition. It's fine like this then.